### PR TITLE
add pin/unpin command

### LINKED ISSUE #126: Implement version pinning
It would be great if we could pin versions for apps such as terraform, kubectl or istioctl, where often, it is not desirable to install the latest version.

### PR DESCRIPTION
**Pull request by snwzd**
closes #126

### CODE REVIEW

**Comment by snwzt** [comment]
I think pin/unpin is done for binaries, please tell me if there is anything that I should change.

**Comment by mbevc1** [comment]
Probably related//solving to #177 too :thinking: ?

File: pkg/config/config.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ type Binary struct {
 	// the package path in config so we don't ask the user to select
 	// the path again when upgrading
 	PackagePath string `json:"package_path"`
+	Pin         bool   `json:"pin"`
```
Comment (marcosnils) [inline pkg/config/config.go:38]: nit: rename to Pinned?

**Comment by marcosnils** [comment]
@snwzt I was coming back to `bin` after a while and I'd like to merge this. Any chance you could rebase? I'll commit to merge it afterwards!

File: cmd/pin.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		Short:         "Pins current version of the binaries",
+		SilenceUsage:  true,
+		Args:          cobra.MinimumNArgs(1),
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg := config.Get()
+
+			binaryList := make(map[string]*config.Binary)
+			for _, b := range cfg.Bins {
+				binaryList[b.RemoteName] = b
```
Comment (marcosnils) [inline cmd/pin.go:30]: This should use `getBinPath`  similar to update here https://github.com/marcosnils/bin/blob/91ae3e63bc1e2a499a832ba5f4e44d3c071b1345/cmd/update.go?plain=1#L58 so `pin` can be called either by the binary name or the path.

File: cmd/unpin.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		Short:         "Unpins current version of the binaries",
+		SilenceUsage:  true,
+		Args:          cobra.MinimumNArgs(1),
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg := config.Get()
+
+			binaryList := make(map[string]*config.Binary)
+			for _, b := range cfg.Bins {
+				binaryList[b.RemoteName] = b
```
Comment (marcosnils) [inline cmd/unpin.go:30]: same as: https://github.com/marcosnils/bin/pull/180/files#r1694289553

File: cmd/update.go [hunk 4]
```diff
@@ -59,6 +59,9 @@ func newUpdateCmd() *updateCmd {
 					if err != nil {
 						return err
 					}
+					if cfg.Bins[bin].Pin {
```
Comment (marcosnils) [inline cmd/update.go:62]: Print an info message here instead of returning an error is better as otherwise, this will interrupt the whole update process.

**Comment by snwzd** [comment]
Hi, I’ve rebased the branch and made the requested changes. Please tell me if there's anything that I should change.

**Pull request merged**